stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelStateLogicImpl.kt

```diff
@@ -247,7 +247,7 @@ internal class ChannelStateLogicImpl(
      * @param message The message to be added or updated.
      */
     override fun upsertMessage(message: Message) {
-        if (mutableState.rawMessages.containsKey(message.id) || mutableState.endOfNewerMessages.value) {
+        if (!mutableState.insideSearch.value) {
```

Comment:
```suggestion
        if (mutableState.rawMessages.containsKey(message.id) || !mutableState.insideSearch.value) {
```

Reply:
We need the contains check to propagate updates to items that are currently inside the list